mungegithub/mungers/submit-queue_test.go

```diff
@@ -746,6 +853,42 @@ func TestSubmitQueue(t *testing.T) {
 				test.pr.Mergeable = nil
 			}
 		})
+		path = "/bar/latest-build.txt"
+		mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+			if r.Method != "GET" {
+				t.Errorf("Unexpected method: %s", r.Method)
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte(strconv.Itoa(test.lastBuildNumber)))
+		})
+		for buildNumber := range test.weakResults {
+			path = fmt.Sprintf("/bar/%v/finished.json", buildNumber)
+			// workaround go for loop semantics
+			buildNumberCopy := buildNumber
+			mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+				if r.Method != "GET" {
+					t.Errorf("Unexpected method: %s", r.Method)
+				}
+				w.WriteHeader(http.StatusOK)
+				data, err := json.Marshal(test.weakResults[buildNumberCopy])
+				if err != nil {
+					t.Errorf("Unexpected error: %v", err)
+				}
+				w.Write(data)
+			})
+		}
+		for junitFile, xml := range test.gcsJunit {
+			path = fmt.Sprintf("/bar/%v/artifacts/%v", test.lastBuildNumber, junitFile)
+			// workaround go for loop semantics
```

Comment:
It's not really a copy. It's one of mine biggest issue with go. When you iterate over something, i.e. write `for x := range bla` compiler allocates a space for a single copy of element of `bla` and reuses it in all iterations. This is OK, for 'normal' processing, but it's kind of painful when you write a go routine inside a for loop or, like here, create a function to be called later.